test/functional/test_framework/messages.py

```diff
@@ -1871,8 +1926,55 @@ def __repr__(self):
 
 import blake3 as BLAKE3
 
+def hex_reverse(h):
+    return "".join(reversed([h[i:i+2] for i in range(0, len(h), 2)]))
+
+class Hash:
+    __slots__ = ("val")
+
+    def __init__(self, val = 0):
+        self.val = val
+        
+    @classmethod
+    def from_hex(cls, hex_str):
+        return cls(int(hex_str, 16))
+
+    @classmethod
+    def from_rev_hex(cls, hex_str):
+        return cls(int(hex_reverse(hex_str), 16))
+        
+    @classmethod
+    def from_byte_arr(cls, b):
+        return cls(deser_uint256(BytesIO(b)))
+    
+    def to_hex(self):
+        return self.serialize().hex()
+
+    def to_byte_arr(self):
+        return hex_str_to_bytes(self.to_hex())
```

Comment:
self.serialize() doesn't work?

Reply:
Yea, that seems to work. I don't actually know what type any of these objects are. Languages without type safety are pure chaos. Everytime I have to write javascript, it feels like I'm fumbling my way through a game of cricket (https://www.youtube.com/watch?v=tYrue4oXCbo)